extensions/arc/src/ui/dashboards/postgres/postgresComputeAndStoragePage.ts

```diff
@@ -483,8 +461,30 @@ export class PostgresComputeAndStoragePage extends DashboardPage {
 	}
 
 	private handleServiceUpdated() {
-		this.editWorkerNodeCount();
-		this.editCores();
-		this.editMemory();
+		if (this._postgresModel.configLastUpdated) {
+			this.editWorkerNodeCount();
+			this.editCores();
+			this.editMemory();
+
+			// Workaround https://github.com/microsoft/azuredatastudio/issues/13134
+			// by only adding these once the model has data. After the bug is fixed,
+			// use loading indicators instead of keeping the page blank.
+			if (this.workerContainer?.items.length === 0) {
```

Comment:
My only concern is duplicated UI components if there are concurrent refreshes that cause a race here.  Doing one refresh as a time in the model is the best fix I know.  But there's a race there too between checking and setting the promise.  @Charles-Gagnon do you know of any better locking strategy in JS?

Reply:
The javascript engine is single threaded - so as long as you don't have anything that'll cause the thread to suspend/switch context (such as awaits) then this should be fine.